examples/heterogeneous-node-classification-with-hashgnn.ipynb

```diff
@@ -0,0 +1,534 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8bccd326",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add logistic regression as a candidate ML algorithm for the training\n",
+    "# Provide an interval for the `penalty` parameter to enable autotuning for it\n",
+    "_ = pipe.addLogisticRegression(penalty=(0.1, 1.0), maxEpochs=1000, patience=5, tolerance=0.0001, learningRate=0.01)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "62fca4ac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add random forest as a candidate ML algorithm for the training\n",
+    "# Provide an interval for the `minSplitSize` parameter to enable autotuning for it\n",
+    "_ = pipe.addRandomForest(minSplitSize=(2, 100), criterion=\"ENTROPY\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "23601d62",
+   "metadata": {},
+   "source": [
+    "## Training the pipeline\n",
+    "\n",
+    "The configuration is done, and we are now ready to kick off the training of our pipeline and see what results we get.\n",
+    "\n",
+    "In our training call, we provide what node label and property we want the training to target, as well as the metric that will determine how the best model candidate is chosen."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "89b8ad98",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Call train on our pipeline object to run the entire training pipeline and produce a model\n",
+    "model, _ = pipe.train(\n",
+    "    G,\n",
+    "    modelName=\"genre-predictor-model\",\n",
+    "    targetNodeLabels=[\"Movie\"],\n",
+    "    targetProperty=\"genre\",\n",
+    "    metrics=[\"ACCURACY\"],\n",
+    "    randomSeed=42,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22cba672",
+   "metadata": {},
+   "source": [
+    "Let's inspect the model that was created by the training pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c9a4f264",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Accuracy scores of trained model:\\n{model.metrics()['ACCURACY']}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "df73b21c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Winning ML algorithm candidate config:\\n{model.best_parameters()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b370419",
+   "metadata": {},
+   "source": [
+    "As we can see the best ML algorithm configuration that the autotuning found was logistic regression with `penalty=0.631876`.\n",
```

Comment:
can you use string interpolation here in case the penalty and scores change later on ?

Reply:
I wouldn't know how to get Python output into markdown like that :O